### PR TITLE
Create troubleshooting.md

### PR DESCRIPTION
**Pull request by oojo12**
# What does this PR do?

Creates a troubleshooting page for the redesigned BYOC page. Currently, we do not have page tailored to address common issues that may arise for users. This PR aims to be the starting point to addressing that.

### CODE REVIEW

**Comment by oojo12** [comment]
@nmiculinic I moved your STS content here to a dedicated troubleshooting page.

**Comment by oojo12** [comment]
A separate PR will be made to remove duplicate content and link all the pages once they have been created.

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Skipping this step will cause difficult to debug issues. The kubelet will be unable to authenticate against the kubernetes API server, and nothing will work.
+
+## Deploying Workloads 
+
+### Stuck in Pending State
+There are usually three scenarios for why a session or run will be stuck in a pending state: awaiting AWS provisioning, one time start-up process **(sessions only)**, or the instance type is not available in the availability zones your BYOC cluster has access to. We will cover each of the three scenarios in detail here.
+
+#### Sessions One Time Startup Process
+
+When ever a BYOC is deployed there is a one time startup process it must go through before it can deploy sessions. Regardless of the instance type selected this process can take up to 30 minutes. Once completed your BYOC will never have to undergo the startup process again before deploying sessions
```
Comment (nmiculinic) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:18]: We handle this by not marking cluster as ready until this process completes, see https://linear.app/gridai/issue/GRID-8362/add-start-up-probe-to-the-gridlet-until-the-snapshot-is-not-ready

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#### Awaiting AWS Provisioning
+
+BYOC will request instances from AWS. Once the isntances have been requested you will have to wait for AWS to grant access to the resources. In Customer-managed BYOC mode you can easily affecct the provisioning time by overprovisioning the instance types of interest. This will effectiviley put the instances on standby, thus increasing costs. However, by overprovisioning instances you will not have to wait for AWS to provision the instances for you as they are requested during BYOC creation.
```
Comment (nmiculinic) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:22]: Grant access to the resource...well, I'd rephrase that to wait for AWS to provision the resources

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+#### Requested Instances not Wihtin Your Availability Zones
+
+The instances requested are AWS instances and thus you will be limited to all AWS instance provisioning limitations. One such limitation is that you will not be able to get instances that are not within your availability zones. There are two ways around this assuming the instances are available within the deployed region:
+
+1. Run `grid edit cluster <cluster name>` and update the "xMultiAzCount" field
+2. If using **Customer-managed BYOC mode** increase the "multi_az_count" variable in <your.tfvars>
```
Comment (nmiculinic) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:29]: We want to deprecate this btw, so this is just temporary, until we disable this completely and use all AZs

Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:29]: gotcha sounds good.

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 2 above)
Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:22]: will do.

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 1 above)
Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:18]: I see that this ticket is in review stage and the associated PR is in draft stage. Once it is merged and released I can update this content to reflect that.

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 2 above)
Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md]: addressed.

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 1 above)
Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:18]: Or @nmiculinic is this expected to go into the next release? I want to keep these docs current with the current release

Comment (essiequoi) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:18]: @oojo12 how will you keep track to make sure to update this when the linear ticket is merged?

**Comment by oojo12** [comment]
I get phone notifications when it things are merged. But I do go through
and do a cursory look at my doc tickets in the morning. So I should notice
it per my normal practice.

**Comment by oojo12** [comment]
Alternatively I can just create one PR with all these changes if that is
your preference.

**Comment by essiequoi** [comment]
@oojo12 ping me when ready for product review

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 1 above)
Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:18]: Engineering will have to inform me.

Comment (nmiculinic) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:18]: No, this will not be in next release, it'll be in one after that

Comment (nmiculinic) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:18]: Just note it'll be released soon, since the fix is already in master

**Review by nmiculinic — APPROVED** [review]

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Bring Your Own Cluster (BYOC) Troubleshooting
+
+## Deploying BYOC
+
+### Kubernetes API Authentication Errors
+AWS STS regional endpoints have to be enabled in the target region. Go to https://console.aws.amazon.com/iam/home#/account_settings and verify the regional endpoint is activated. In most cases your region already has AWS STS regional endpoint enabled -> https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html
```
Comment (essiequoi) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:6]: 1. What happens if endpoints are not enabled? Could you rephrase this to be in this format: In order to do [a thing], AWS STS regional endpoints have to be enabled in the target region. 

2. Create a text hyperlink instead of just pasting the link. So you can update that line to 
Go to your [Amazon account settings](https://console.aws.amazon.com/iam/home#/account_settings) and verify...

3. add: "that" (...and verify **that** the regional endpoint is activated).

Comment (essiequoi) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:6]: 4. In most cases your region already has AWS STS regional endpoint enabled -> 
Can you specify in a hyper-linked sentence what this link is pointing to?

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+!!! note
```
Comment (essiequoi) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:8]: add this line to the top of the page: `import Note from "@site/src/components/Note";`
and then wrap the note in `<note></note>` instead of using the !!! note syntax

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    Skipping this step will cause difficult to debug issues. The kubelet will be unable to authenticate against the kubernetes API server, and nothing will work.
```
Comment (essiequoi) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:9]: Skipping this step will cause issues which are difficult to debug. The kubelet will be unable to authenticate against the Kubernetes API server, breaking core functionality.

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+## Deploying Workloads 
+
+### Stuck in Pending State
+There are usually three scenarios for why a session or run will be stuck in a pending state: awaiting AWS provisioning, one time start-up process **(sessions only)**, or the instance type is not available in the availability zones your BYOC cluster has access to. We will cover each of the three scenarios in detail here.
```
Comment (essiequoi) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:14]: Use numbered list when listing things numerically. Capitalize Session and Run. 

There are usually three scenarios for why a Session or Run will be stuck in a pending state: 
1. Awaiting AWS provisioning
2. For Sessions only: Initial start-up process (only happens once)
3. The requested Instance type is not available in the availability zones your cluster has access to

We will cover each of the three scenarios in detail here.

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 1 above)
Comment (essiequoi) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:18]: Let's not call it a BYOC since that might be ambiguous to users and we haven't really defined it anywhere. Let's say:

Whenever a new cluster is deployed, there is a one-time startup process...

File: docs/platform/custom-cloud-credentials/troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+#### Sessions One Time Startup Process
+
+When ever a BYOC is deployed there is a one time startup process it must go through before it can deploy sessions. Regardless of the instance type selected this process can take up to 30 minutes. Once completed your BYOC will never have to undergo the startup process again before deploying sessions
+
+#### Awaiting AWS Provisioning
+
+BYOC will request instances from AWS. Once the isntances have been requested you will have to wait for AWS to provison the resources. In Customer-managed BYOC mode you can easily affecct the provisioning time by overprovisioning the instance types of interest. This will effectiviley put the instances on standby, thus increasing costs. However, by overprovisioning instances you will not have to wait for AWS to provision the instances for you as they are requested during BYOC creation.
```
Comment (essiequoi) [inline docs/platform/custom-cloud-credentials/troubleshooting.md:22]: typo: instances (Once the instances have been requested...)

**Review by essiequoi — CHANGES_REQUESTED** [review]
Hey @oojo12 added some comments. After this, I think we are good to go.

File: docs/platform/custom-cloud-credentials/troubleshooting.md (hunk 1 above)
Comment (oojo12) [reply docs/platform/custom-cloud-credentials/troubleshooting.md:26]: BYOC is defined at the top of the page
